src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -401,7 +399,53 @@ void consumeRecords(final List<O> recOps) {
     }
   }
 
-  void createTopics(final List<O> topicOps) {}
+  void createTopics(final List<O> topicOps) {
+    val nodeAddr = topicOps.get(0).nodeAddr();
+    try {
+      val config = configCache.computeIfAbsent(nodeAddr, this::createAdminClientConfig);
+      val adminClientCreateFunc =
+          adminClientCreateFuncCache.computeIfAbsent(config, AdminClientCreateFunctionImpl::new);
+      val adminClient = adminClientCache.computeIfAbsent(nodeAddr, adminClientCreateFunc);
+      final Collection<NewTopic> topicCollection = new ArrayList<>();
+      for (var i = 0; i < topicOps.size(); i++) {
+        val topicOp = topicOps.get(i);
+        val topicName = topicOp.item().name();
+        val newTopic = new NewTopic(topicName, 1, (short) 1);
+        topicCollection.add(newTopic);
+      }
+      concurrencyThrottle.acquire();
+      val createTopicsResult = adminClient.createTopics(topicCollection);
+      for (var i = 0; i < topicOps.size(); i++) {
+        val topicOp = topicOps.get(i);
+        val topicName = topicOp.item().name();
+        try {
+          topicOp.startRequest();
+          val oneTopicResult = createTopicsResult.values().get(topicName);
+          oneTopicResult.whenComplete(
+              ((aVoid, throwable) -> {
+                if (throwable == null) {
+                  completeOperation((O) topicOp, SUCC);
+                } else {
+                  LogUtil.exception(
+                      Level.DEBUG,
+                      throwable,
+                      "{}: Failed to create topic \"{}\"",
+                      stepId,
+                      topicName);
+                }
+              }));
+          topicOp.finishRequest();
+        } catch (final TopicExistsException e) {
+          LogUtil.exception(Level.DEBUG, e, "{}: Topic \"{}\" already exists", stepId, topicName);
+          completeOperation((O) topicOp, RESP_FAIL_UNKNOWN);
+        }
+      }
+    } catch (final Throwable thrown) {
+      throwUncheckedIfInterrupted(thrown);
+    } finally {
+      concurrencyThrottle.release();
```

Comment:
not safe: this will cause concurrency inconsistency if the exception is thrown before the `concurrencyThrottle.acquire()` call
let's introduce inner try-finally block right after the `concurrencyThrottle.acquire()` call